symengine/CMakeLists.txt

```diff
@@ -218,6 +218,10 @@ if (NO_WARN_IMPLICIT_FALL_THROUGH_FLAG)
     set_source_files_properties(parser/scanner.cpp PROPERTIES COMPILE_FLAGS "-Wno-conversion -Wno-implicit-fallthrough")
 endif()
 
+if (WITH_LLVM AND EXCLUDE_LIBS AND NOT SYMENGINE_LLVM_SHARED)
+    set_property(TARGET symengine APPEND_STRING PROPERTY LINK_FLAGS "-Wl,--exclude-libs,ALL")
```

Comment:
I merged it, so that we can use the feature, but would you please mind documenting here (as a comment in the source code) why we need the `-Wl,--exclude-libs,ALL` linker flags? I already forgot the discussion about it.